tapdb/sqlc/migrations/000020_asset_unique_key.up.sql

```diff
@@ -1,3 +1,43 @@
+-- First step is to make sure all duplicate assets are seen as spent, if their
+-- anchor transaction is spent. That way we can later also group on the spent
+-- column to remove duplicates.
+UPDATE assets
+SET spent = true
+WHERE asset_id IN (SELECT a.asset_id
+                   FROM assets a
+                            JOIN managed_utxos mu
+                                 ON a.anchor_utxo_id = mu.utxo_id
+                            JOIN chain_txns ct
+                                 ON mu.txn_id = ct.txn_id
+                            LEFT JOIN asset_transfer_inputs ati
+                                      ON ati.anchor_point = mu.outpoint
+                   WHERE a.spent = false
+                     AND ati.input_id IS NOT NULL);
+
+-- Now remove all asset witnesses for the assets we're going to remove in the
+-- next step.
+DELETE
```

Comment:
Can we safely delete the witness of a duplicate asset? Are we sure the witness is also available for the asset that will remain? Otherwise we should UPDATE the foreign key of the asset witness.

Reply:
Yes, I think so. We always insert the witness with the asset ID that was inserted in the previous step. So every asset should have a witness.

Reply:
Update the logic to make sure we fill any gaps for duplicates that for some reason didn't have a witness.